deploy-docker/action.yml

```diff
@@ -0,0 +1,24 @@
+name: 'Deploy Docker'
+description: Deploys the build to docker
+inputs:
+  component:
+    description: 'Component name to build the image for'
+    required: true
+  docker-user:
+    description: Username of the user to use when logging in to docker to deploy
+    required: true
+  docker-password:
+    description: Password of the docker user
+    required: true
+  private-key:
```

Comment:
We must add a very big fat warning somewhere to use multi-stage builds in order to prevent leaving behind the keys in a (public) Docker image. See e.g. https://vsupalov.com/build-docker-image-clone-private-repo-ssh-key/

Reply:
the private key is only used by some golang images as they are the only ones that build things themselves and not copy over the executable. So we could make all of them also just copy over and remove this option altogether, but it is already so that those that do build it themselves over use multi-stage builds

Reply:
If easy enough to do, let's stop passing the private key to the docker build process altogether to prevent a potential leak of a very sensitive secret.